### PR TITLE
Add 3.5 versions

### PR DESCRIPTION
**Pull request by odersky**

### CODE REVIEW

**Review by nicolasstucki — APPROVED** [review]

**Comment by nicolasstucki** [comment]
We might need to add this to 3.4.0. This implies that we might need a 3.4.0-RC2.

**Comment by bishabosha** [comment]
why would you need a 3.5 language import in 3.4?

File: library/src/scala/runtime/stdLibPatches/language.scala [hunk 1]
```diff
@@ -249,6 +249,20 @@ object language:
...
+
+  /** Set source version to 3.5
+    *
+    * @see [[https://docs.scala-lang.org/scala3/guides/migration/compatibility-intro.html]]
+    */
+  @compileTimeOnly("`3.5` can only be used at compile time in import statements")
+  object `3.5`
+
   // !!! Keep in sync with dotty.tools.dotc.config.SourceVersion !!!
   // Also add tests in `tests/pos/source-import-3-x.scala` and `tests/pos/source-import-3-x-migration.scala`
```
Comment (bishabosha) [inline library/src/scala/runtime/stdLibPatches/language.scala:267]: there's also these test files to update

Comment (odersky) [reply library/src/scala/runtime/stdLibPatches/language.scala:267]: can you take care of that? Thanks!

**Pull request merged**